auth0/resource_auth0_guardian_test.go

```diff
@@ -6,302 +6,293 @@ import (
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
 )
 
+const testAccGuardianEmailCreate = `
+resource "auth0_guardian" "foo" {
+	policy = "all-applications"
+	email  = true
+}
+`
+
+const testAccGuardianEmailDelete = `
+resource "auth0_guardian" "foo" {
+	policy = "all-applications"
+	email  = false
+}
+`
+
+const testAccGuardianOTPCreate = `
+resource "auth0_guardian" "foo" {
+	policy = "all-applications"
+	otp    = true
+}
+`
+
+const testAccGuardianOTPDelete = `
+resource "auth0_guardian" "foo" {
+	policy = "all-applications"
+	otp    = false
+}
+`
+
+const testAccGuardianRecoveryCodeCreate = `
+resource "auth0_guardian" "foo" {
+	policy        = "all-applications"
+	recovery_code = true
+}
+`
+
+const testAccGuardianRecoveryCodeDelete = `
+resource "auth0_guardian" "foo" {
+	policy        = "all-applications"
+	recovery_code = false
+}
+`
+
 func TestAccGuardian(t *testing.T) {
 	httpRecorder := configureHTTPRecorder(t)
 
 	resource.Test(t, resource.TestCase{
 		ProviderFactories: testProviders(httpRecorder),
 		Steps: []resource.TestStep{
 			{
-				Config: testAccConfigureTwilio,
-				Check: resource.ComposeTestCheckFunc(
-					resource.TestCheckResourceAttr("auth0_guardian.foo", "policy", "all-applications"),
-					resource.TestCheckResourceAttr("auth0_guardian.foo", "phone.0.message_types.0", "sms"),
-					resource.TestCheckResourceAttr("auth0_guardian.foo", "phone.0.provider", "twilio"),
-					resource.TestCheckResourceAttr("auth0_guardian.foo", "phone.0.options.0.enrollment_message", "enroll foo"),
-					resource.TestCheckResourceAttr("auth0_guardian.foo", "phone.0.options.0.verification_message", "verify foo"),
-					resource.TestCheckResourceAttr("auth0_guardian.foo", "phone.0.options.0.from", "from bar"),
-					resource.TestCheckResourceAttr("auth0_guardian.foo", "phone.0.options.0.messaging_service_sid", "foo"),
-					resource.TestCheckResourceAttr("auth0_guardian.foo", "phone.0.options.0.auth_token", "bar"),
-				),
-			},
-			{
-				Config: testAccConfigureTwilioUpdate,
+				Config: testAccGuardianEmailCreate,
 				Check: resource.ComposeTestCheckFunc(
 					resource.TestCheckResourceAttr("auth0_guardian.foo", "policy", "all-applications"),
+					resource.TestCheckResourceAttr("auth0_guardian.foo", "email", "true"),
+					resource.TestCheckResourceAttr("auth0_guardian.foo", "otp", "false"),
 					resource.TestCheckResourceAttr("auth0_guardian.foo", "phone.#", "0"),
+					resource.TestCheckResourceAttr("auth0_guardian.foo", "duo.#", "0"),
+					resource.TestCheckResourceAttr("auth0_guardian.foo", "webauthn_platform.#", "0"),
+					resource.TestCheckResourceAttr("auth0_guardian.foo", "webauthn_roaming.#", "0"),
+					resource.TestCheckResourceAttr("auth0_guardian.foo", "push.#", "0"),
+					resource.TestCheckResourceAttr("auth0_guardian.foo", "recovery_code", "false"),
```

Comment:
We're splitting the 1 big test into multiple ones that focus on distinct MFA types. Also we're not checking the full state for all the possible parameters for a thorough inspection.